lib/easy_broker/query.rb

```diff
@@ -2,16 +2,23 @@
 
 class EasyBroker::Query
   attr_reader :api_client, :endpoint, :query_params
+  DEFAULT_PAGE = 1
 
   def initialize(api_client, endpoint, query_params)
     @api_client = api_client
     @endpoint = endpoint
     @query_params = query_params
   end
 
-  def get(page = 1)
-    query_params[:page] = page
+  def get(page = nil)
+    query_params[:page] = page_param(page)
```

Comment:
@enortham Hugo helped me with a bug for the `page` param, it was always set as `1`, we implemented the `page_param` method to select from `page` (so `next_page` can work), `query_params[:page]` or `DEFAULT_PAGE`. What do you think about this approach?

Reply:
Sorry, I don't follow. Why didn't passing the param work before? I'm almost certain that I tested this live several times but I also don't see why it wouldn't override the param on every call to `get`.

Reply:
@enortham We tested in `master` and `next_page` seems not working as expected, every time we use it, it stays in the same page
E.g. when using this:
```ruby
client.properties.search(query: { search: { statuses: [:published] }, page: 3, limit: 1 })
```
The query built is this:
```ruby
{:query=>
  {:query=>{:search=>{:statuses=>[:published]}, :page=>3, :limit=>1},
   :page=>1}}
```
As you can see, because `page` is always `1` in the `get` method, `query_params[:page]`  send `page` as `1`, but it's ignored for HTTParty.
Now if we try to use `next_page` as follows:
```ruby
results = client.properties.search(query: { search: { statuses: [:published] }, page: 3, limit: 1 })
results.next_page
```
The params to get the next page are these:
```ruby
{:query=>
  {:query=>{:search=>{:statuses=>[:published]}, :page=>3, :limit=>1},
   :page=>4}}
```
`page` outside `query` is correctly set, but the actual used by HTTParty is `3`, so it will not show the next page. The issue is more evident with the changes in this branch, because `page` and `limit` are not wrapped with `query`, when you try to use `search` like this:
```ruby
client.properties.search(query: { search: { statuses: [:published] }, page: 3, limit: 1 })
```
`page => 1` remains no matter what.

Reply:
Hmm, can you just fix it with the following two line change since nil should default to 1? 
```ruby
def get(page = nil)
    query_params[:page] = page
```

Reply:
I'm pretty sure that if the page is nil it returns page 1. You could also change it to  `= page if page` if you want to return the original in set when it's instantiated.